features/docs/gegeven-stap-definities-adoptie.feature

```diff
@@ -0,0 +1,60 @@
+
+  @stap-documentatie
+  Functionaliteit: Persoon is geadopteerd
+
+  Achtergrond:
+  Gegeven de 1e 'SELECT COALESCE(MAX(pl_id), 0)+1 FROM public.lo3_pl' statement heeft als resultaat '9999'
+  En de 2e 'SELECT COALESCE(MAX(pl_id), 0)+1 FROM public.lo3_pl' statement heeft als resultaat '10000'
+  En de 3e 'SELECT COALESCE(MAX(pl_id), 0)+1 FROM public.lo3_pl' statement heeft als resultaat '10001'
+
+  Scenario: is geadopteerd door persoon als ouder 1
+    Gegeven de persoon 'P2' met burgerservicenummer '000000012'
+    En de persoon 'P1' met burgerservicenummer '000000036'
+    * is minderjarig
+    * is geadopteerd door 'P2' als ouder 1
+      Dan zijn de gegenereerde SQL statements
+      | stap | categorie    | text                                                                                                                                                                     | values                                              |
+      | 1    | inschrijving | INSERT INTO public.lo3_pl(pl_id,mutatie_dt,geheim_ind) VALUES((SELECT COALESCE(MAX(pl_id), 0)+1 FROM public.lo3_pl),current_timestamp,$1) RETURNING *                    | 0                                                   |
+      |      | persoon      | INSERT INTO public.lo3_pl_persoon(pl_id,stapel_nr,volg_nr,persoon_type,burger_service_nr,geslachts_naam) VALUES($1,$2,$3,$4,$5,$6)                                       | 9999,0,0,P,000000012,P2                             |
+      |      | kind-1       | INSERT INTO public.lo3_pl_persoon(pl_id,stapel_nr,volg_nr,persoon_type,burger_service_nr,geslachts_naam) VALUES($1,$2,$3,$4,$5,$6)                                       | 9999,0,0,K,000000036,P1                             |
+      | 2    | inschrijving | INSERT INTO public.lo3_pl(pl_id,mutatie_dt,geheim_ind) VALUES((SELECT COALESCE(MAX(pl_id), 0)+1 FROM public.lo3_pl),current_timestamp,$1) RETURNING *                    | 0                                                   |
+      |      | persoon      | INSERT INTO public.lo3_pl_persoon(pl_id,stapel_nr,volg_nr,persoon_type,burger_service_nr,geslachts_naam,geboorte_datum,akte_nr) VALUES($1,$2,$3,$4,$5,$6,$7,$8)          | 10000,0,0,P,000000036,P1,gisteren - 17 jaar,1AQ0100 |
+      |      | persoon      | INSERT INTO public.lo3_pl_persoon(pl_id,stapel_nr,volg_nr,persoon_type,burger_service_nr,geslachts_naam,geboorte_datum) VALUES($1,$2,$3,$4,$5,$6,$7)                     | 10000,0,1,P,000000036,P1,gisteren - 17 jaar         |
+      |      | ouder-1-1    | INSERT INTO public.lo3_pl_persoon(pl_id,stapel_nr,volg_nr,persoon_type,burger_service_nr,geslachts_naam,familie_betrek_start_datum,akte_nr) VALUES($1,$2,$3,$4,$5,$6,$7) | 10000,0,0,1,000000012,P2,gisteren - 4 jaar,1AQ0100  |
+
+  Scenario: is geadopteerd door persoon als ouder 2
+    Gegeven de persoon 'P2' met burgerservicenummer '000000012'
+    En de persoon 'P1' met burgerservicenummer '000000036'
+    * is minderjarig
+    * is geadopteerd door 'P2' als ouder 2
```

Comment:
Wanneer een ouder een 'normale' ouder is, voeren we die nu op met de stap ```heeft een ouder 1 'P1' met burgerservicenummer '000000012'```.
Maar wanneer de ouder een adoptie ouder is, gebruiken we twee stappen: eerst ```de persoon 'P1' met burgerservicenummer '000000012'``` en vervolgens ```is geadopteerd door 'P1' als ouder 1```.

Dat is niet handig denk ik. Beter om één methode te kiezen. Ik denk dat dan de 2-staps methode flexibeler bruikbaar is.

Reply:
Dus dat we een stap ```heeft een ouder 1 'P1'``` krijgen. Die maakt dan de ouder relatie vanuit de pl van de minderjarige persoon én een kind relatie vanuit meerderjarige P1. In dat geval vervangt dat stap ```heeft een ouder 1 'P1' met burgerservicenummer '000000012'`` (die kan dan vervallen)

Reply:
En ```heeft een ouder '(\d)' met de volgende gegevens``` kan dan denk ik ook vervallen. De stap ```heeft een ouder 1 'P1'``` is handiger, want die legt (als het goed is) zowel de ouder-relatie als de kind-relatie aan.